jac/jaclang/compiler/parser.py

```diff
@@ -601,8 +601,9 @@ def arch_type(self, kid: list[ast.AstNode]) -> ast.Token:
                     | KW_EDGE
                     | KW_NODE
             """
-            if isinstance(kid[0], ast.Token):
-                return kid[0]
+            if arch_token := self.match(ast.Token):
+                return arch_token
+
             else:
                 raise self.ice()
 
```

Comment:
Same here, remove `else: raise self.ice()`, and `return self.consume(ast.Token)`